Add a Score counter to task overview and sidebar

pxsit opened the pull request:
Add a Score counter to task overview
<img width="923" height="430" alt="image" src="https://github.com/user-attachments/assets/18d141e5-73f2-4ac5-8627-88af9f19468f" />

pxsit committed:
Add a Score counter to task overview

Add a Score counter to task overview

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

Muaath5 commented:
> I have some comments about the code, but i'm wondering if this is the UI we want in the first place... I saw some browser extension that showed the task scores on the sidebar next to the task links, maybe that would be better.
> 
> Does anyone else have any thoughts?

I don't see any issues if both of these ways where implemented, but if it was in the task overview it would be better in case of onsite competitions and you don't want others to look into your score :)

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Restore Formatting, Make it toggleable

pxsit committed:
Fix color not displaying

pxsit commented:
> > I have some comments about the code, but i'm wondering if this is the UI we want in the first place... I saw some browser extension that showed the task scores on the sidebar next to the task links, maybe that would be better.
> > Does anyone else have any thoughts?
> 
> I don't see any issues if both of these ways where implemented, but if it was in the task overview it would be better in case of onsite competitions and you don't want others to look into your score :)

Added an option to toggle this in AWS (:

pxsit committed:
Fix

Reformat, Fix Updaters, more

prandla commented on cms/server/contest/templates/overview.html:
```diff
@@ -208,6 +211,9 @@ <h2>{% trans %}Task overview{% endtrans %}</h2>
 {% for t_iter in contest.tasks %}
         <tr>
+{% if contest.show_task_scores_in_overview and task_scores is defined %}
+            <td class="public_score {{ get_score_class(task_scores[t_iter.id][0], task_scores[t_iter.id][1]) }}">{{ task_scores[t_iter.id][2] }}</td>
```
Please test your changes locally before pushing next time. This crashes, `get_score_class` (the python function, not the javascript function with the same name...) takes 3 arguments, the 3rd one is the score precision.

prandla commented on cms/server/contest/static/cws_style.css:
```diff
@@ -559,27 +559,27 @@ td.token_rules p:last-child {
 }
 
-.submission_list td.public_score.score_0 {
+.main_task_list td.public_score.score_0 {
```
You broke the coloring in the submission list of a single task with this. I meant you should change the entire line to just `td.public_score.score_0 {`.

prandla commented on cms/server/contest/handlers/main.py:
```diff
@@ -381,7 +438,7 @@ def get(self):
         if config.docs_path is not None:
             for language in languages:
-                ext = language.source_extensions[0][1:]  # remove dot
+                ext = language.source_extensions[0][1:] # remove dot
```
sorry, i didn't notice this earlier: please undo this formatting change. 1) it's unrelated to this PR (in fact, it causes a conflict with another pending PR) and 2) it's not what `ruff format` does.

pxsit committed:
Fix the Fix

pxsit committed:
Fix the Fix that Fix the Fix

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Update cws_style.css

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Add credit for original owner?

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Update update_from_1.5.sql

Looks cursed having it not in order

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'cms-dev:main' into add-counter

pxsit committed:
Merge branch 'cms-dev:main' into add-counter

pxsit committed:
Merge branch 'cms-dev:main' into add-counter

pxsit committed:
Merge branch 'cms-dev:main' into add-counter

pxsit committed:
Remove unused PrintJob import from main.py

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Optimize contest data loading

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Add sidebar task scores for contests

Introduce an option to display per-task public scores in the contest sidebar. Adds a new DB column (show_task_scores_in_sidebar) and admin checkbox to toggle it, includes CSS badge styling and template changes to render the score next to each task. Backend changes add helpers to load participations with the relations required for scoring and to compute formatted public task scores (only computed when needed and hiding tasks with zero public max in the sidebar). Includes SQL updater to migrate existing databases.

pxsit committed:
Update update_from_1.5.sql

pxsit committed:
Show tokened task scores in sidebar

Add logic to display tokened/total task scores in the sidebar when a token has been played on a task or when in analysis phase. Introduce _compute_sidebar_task_scores in the contest handler and switch render_params to use it (passing actual_phase) so sidebar values use tokened scores where appropriate while still hiding tasks with no public score. Templates updated: add data-task-name attribute to task nav headers and add JS to initialize sidebar state, update sidebar badges dynamically (sidebar_use_tokened_score, update_sidebar_task_score) and switch displayed score to tokened totals when submissions reveal full scores. This enables correct initial rendering and real-time updates of sidebar task scores based on token usage and scoring visibility.

pxsit committed:
Enable task scores in sidebar by default

Change default for show_task_scores_in_sidebar from False to True in cms/db/contest.py so new contests display task scores in the sidebar by default. Also remove an extraneous blank line from the SQL updater script (formatting cleanup).

pxsit committed:
Unify task score computation into _compute_task_scores

Replace separate _compute_public_task_scores and sidebar-specific logic with a single _compute_task_scores method that handles both public and tokened/total scores. The new method accepts actual_phase and a hide_zero_max_public flag (default true) to control whether tasks with zero public max score are shown. Update callers in contest/handlers/contest.py and contest/handlers/main.py to use the unified method and pass appropriate hide_zero_max_public values, preserving previous visibility behavior.

pxsit committed:
Show full scores in analysis mode and minor fixes

When the contest is in analysis mode (actual_phase == 3) total/task tokened scores are now shown as full scores instead of being limited to tokened submissions. Added joinedload(Participation.user) to the contest handler to eager-load users and avoid extra queries, and replaced per-task any(...) scans with a tokened_task_ids set for efficiency. The task submission handler now checks actual_phase to decide whether to restrict to tokened submissions. The template always renders a task score badge placeholder when sidebar scores are enabled so layout remains consistent. Also clarified related docstrings/comments.

pxsit committed:
Reuse preloaded participation and show N/A scores

Avoid a duplicate DB query by reusing a participation object that ContestHandler may have preloaded for sidebar scores; only call _load_participation_for_scores when necessary. Compute task scores conditionally as before. UI fixes: display "N/A" for undefined task scores in the contest sidebar, add CSS for the undefined badge state, and ensure task submission badges remove the undefined class and are shown after updating the score.

pxsit committed:
Update contest.html

pxsit committed:
Update contest.py

pxsit committed:
Added an option to display task scores in the contest sidebar

Add an option to display task scores in the contest sidebar and for both the overview and the sidebard task scores, when a tokened score is availble it will use that score, if not it will use the public score, if there is no public score it will show 0/0 (considering to change to N/A)

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Fix bugs & Optimize

pxsit committed:
Update contest.html

pxsit committed:
Fix bugs

pxsit committed:
Update contest.py

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge upstream e0e4f8b7 commit

pxsit committed:
Fix template get_score_class calls to use new 2-arg signature

pxsit committed:
Merge branch 'main' into add-counter

pxsit committed:
Merge branch 'cms-dev:main' into add-counter
